hazelcast/src/main/java/com/hazelcast/query/impl/OrderedIndexStore.java

```diff
@@ -43,26 +44,27 @@
  */
 @SuppressWarnings("rawtypes")
 public class OrderedIndexStore extends BaseSingleValueIndexStore {
+    private static final DataComparator DATA_COMPARATOR = new DataComparator();
 
-    private final ConcurrentSkipListMap<Comparable, Map<Data, QueryableEntry>> recordMap =
+    private final ConcurrentSkipListMap<Comparable, NavigableMap<Data, QueryableEntry>> recordMap =
```

Comment:
The change may influence the memory consumption as well as queries performance. We need to run some benchmarks to understand how the change influences this.